version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: sistema-pedidos-app
    volumes:
      - .:/var/www/html
      - ./docker/php/php.ini:/usr/local/etc/php/conf.d/custom.ini
    environment:
      DATABASE_URL: "postgresql://symfony_user:senha_segura@database:5432/sistema_pedidos"
    depends_on:
      database:
        condition: service_healthy
    networks:
      - app-network
    expose:
      - 9000

  database:
    image: postgres:15
    container_name: sistema-pedidos-db
    restart: always
    environment:
      POSTGRES_DB: sistema_pedidos
      POSTGRES_USER: symfony_user
      POSTGRES_PASSWORD: senha_segura
      TZ: America/Sao_Paulo
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD", "pg_isready", "-U", "symfony_user", "-d", "sistema_pedidos" ]
      interval: 10s
      retries: 5
      start_period: 30s
      timeout: 10s
    volumes:
      - database_data:/var/lib/postgresql/data
      - ./docker/pgsql/init.sql:/docker-entrypoint-initdb.d/init.sql
    entrypoint: [ "/bin/sh", "-c", "chmod 644 /docker-entrypoint-initdb.d/init.sql && chown postgres:postgres /docker-entrypoint-initdb.d/init.sql && docker-entrypoint.sh postgres" ]
    networks:
      - app-network

  nginx:
    image: nginx:latest
    container_name: sistema-pedidos-nginx
    ports:
      - "8082:80"
    volumes:
      - .:/var/www/html
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - app
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  database_data:
